pkg/controllers/subnet/subnet_controller.go

```diff
@@ -233,6 +273,43 @@ func (r *SubnetReconciler) updateSubnetStatus(obj *v1alpha1.Subnet) error {
 	return nil
 }
 
+// handleSharedSubnet handles a shared subnet with associated_resource label
+// It gets the NSX subnet based on the associated resource (first from cache, then from NSX API),
+// updates the status of the Subnet CR, and adds the subnet to the polling queue for regular status updates.
+func (r *SubnetReconciler) handleSharedSubnet(ctx context.Context, subnetCR *v1alpha1.Subnet, namespacedName client.ObjectKey, associatedResource string) (ctrl.Result, error) {
+	log.Info("Subnet has associated_resource label, skipping NSX Subnet creation", "Subnet", namespacedName, "AssociatedResource", associatedResource)
+
+	// Get NSX subnet from cache or API
+	nsxSubnet, err := r.SubnetService.GetNSXSubnetFromCacheOrAPI(associatedResource, namespacedName)
+	if err != nil {
+		r.updateSharedSubnetWithError(ctx, namespacedName, err, "Failed to get NSX Subnet for associated resource")
+		return ResultRequeue, err
+	}
+
+	// Get subnet status from cache or API
+	statusList, err := r.SubnetService.GetSubnetStatusFromCacheOrAPI(nsxSubnet, associatedResource, namespacedName)
+	if err != nil {
+		// Use updateSharedSubnetWithError for consistency with pollAllSharedSubnets
+		r.updateSharedSubnetWithError(ctx, namespacedName, err, "NSX subnet status")
+		return ResultRequeue, err
+	}
+
+	// Update the cache with the NSX subnet and status list
+	r.SubnetService.UpdateNSXSubnetCache(associatedResource, nsxSubnet, statusList)
```

Comment:
subnet_poll will regularly poll latest NSX subnet and update the cache, here can we just read from cache and do not update cache?

Reply:
yes, fixd